src/EditorFeatures/Core.Wpf/InlineRename/Dashboard/DashboardViewModel.cs

```diff
@@ -271,7 +271,11 @@ public bool DefaultRenameOverloadFlag
                 if (IsRenameOverloadsEditable)
                 {
                     _defaultRenameOverloadFlag = value;
-                    _session.RefreshRenameSessionWithOptionsChanged(RenameOptions.RenameOverloads, value);
+
+                    if (_session.Options.RenameOverloads != value)
+                    {
+                        _session.RefreshRenameSessionWithOptionsChanged(_session.Options with { RenameOverloads = value });
+                    }
```

Comment:
should we just pass in the changed value, and then allow RefreshRenameSessionWithOptionsChanged to itself do nothing if there's no change?

Reply:
It was previously done that way, but with strongly typed option record it's easier to do it this way.